src/Protocol/Emitter.php

```diff
@@ -18,9 +18,9 @@ final class Emitter
 
     private FileWriter $writer;
 
-    public function __construct(string $filename = 'dump.lsif')
+    public function __construct(string $filename = 'dump.lsif', int $id = 0)
     {
-        $this->id = 0;
+        $this->id = $id;
```

Comment:
FYI: The ID is incremented before it is used (see `private function  id(): int` at the end of this file), so `$id + 1` is the first actual ID, which is emitted. I'm sure whether this is the behavior you want, you may want to change that.

Reply:
@davidrjenni I improved this experience here. I think it now aligns with how it should be used, where you actually pass the id you want as the first element id.